uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Form/WorkPattern.php

```diff
@@ -0,0 +1,152 @@
+    {
+        if ($this->_action & (CRM_Core_Action::ADD | CRM_Core_Action::UPDATE)) {
+            // store the submitted values in an array
+            $params = $this->exportValues();
+
+            if ($this->_action & CRM_Core_Action::UPDATE) {
+                $params['id'] = $this->_id;
+            }
+
+            //when a checkbox is not checked, it is not sent on the request
+            //so we check if it wasn't sent and set the param value to 0
+            $checkboxFields = ['is_default', 'is_active'];
+            foreach ($checkboxFields as $field) {
+                if(!array_key_exists($field, $params)) {
+                    $params[$field] = 0;
+                }
+            }
+
+            $actionDescription = ($this->_action & CRM_Core_Action::UPDATE) ? 'updated' : 'created';
+            try {
+                $workPattern = CRM_HRLeaveAndAbsences_BAO_WorkPattern::create($params);
+                CRM_Core_Session::setStatus(ts("The Work Pattern '%1' has been $actionDescription.", array( 1 => $workPattern->label)), 'Success', 'success');
+            } catch(Exception $ex) {
+                $message = ts("The Work Pattern could not be $actionDescription.");
+                $message .= ' ' . $ex->getMessage();
+                CRM_Core_Session::setStatus($message, 'Error', 'error');
+            }
+
+            $url = CRM_Utils_System::url('civicrm/admin/leaveandabsences/work_patterns', 'reset=1&action=browse');
+            $session = CRM_Core_Session::singleton();
+            $session->replaceUserContext($url);
+        }
+    }
+
+    /**
+     * Adds the fields of the Details tab to the form
+     */
+    private function addBasicDetailsFields()
+    {
+        $this->add(
+            'text',
+            'label',
+            ts('Label'),
+            $this->getDAOFieldAttributes('label'),
+            true
+        );
+        $this->add(
+            'text',
+            'description',
+            ts('Description'),
+            $this->getDAOFieldAttributes('description')
+        );
+        $this->add(
+            'checkbox',
+            'is_active',
+            ts('Enabled')
+        );
+        $this->add(
+            'checkbox',
+            'is_default',
+            ts('Is default')
+        );
+    }
+
+    private function getDAOFieldAttributes($field)
+    {
+        $dao = 'CRM_HRLeaveAndAbsences_DAO_WorkPattern';
+        return CRM_Core_DAO::getAttribute($dao, $field);
+    }
+
+    /**
+     * An helper method to retrieve the buttons available
+     * on the form.
+     *
+     * The Delete button is only available while editing an
+     * existing Work Pattern
+     *
+     * @return array
+     */
+    private function getAvailableButtons()
+    {
+        $buttons = [
+            [ 'type' => 'next', 'name' => ts('Save'), 'isDefault' => true ],
+            [ 'type' => 'cancel', 'name' => ts('Cancel') ],
+        ];
+
+        if($this->_action & CRM_Core_Action::UPDATE) {
```

Comment:
shouldn't this be something like this :

`$this->_action == CRM_Core_Action::UPDATE`